ts/store/reducers/entities/services/servicesById.ts

```diff
@@ -39,9 +42,17 @@ const reducer = (
 
     case getType(loadService.failure):
       // Use the ID as object key
+      const { service_id, to_remove } = action.payload;
+      const prevServiceFailure = state[service_id];
+      const nextServiceFailure =
```

Comment:
We have to change the pot status accordly to the event occurred, so the pot should be in error state

```typescript
// if we had to remove this item from the store (404/401) delete the key
// otherwise we have to check if the store has a valid item inside (isSome)
// and set the pot to someError
if (to_remove) {
  const clonedState = { ...state };
  // tslint:disable-next-line: no-object-mutation
  delete clonedState[service_id];
  return clonedState;
}
const nextServiceFailure =
  prevServiceFailure !== undefined && pot.isSome(prevServiceFailure)
    ? pot.someError(prevServiceFailure.value, Error(action.payload.service_id))
    : pot.noneError(Error());
return {
  ...state,
  [service_id]: nextServiceFailure
};

```

Reply:
Thinking deeply about the failure case, the best approach should be to relax the deletion: if the request went successfully and the status code is 404/401, we could move the pot in error state but avoid delete.

This is because if a user has a message where the sender service has been removed, starting from that point he will never know the service details because the app deletes that entry from the cache